packages/edit-site/src/hooks/use-global-styles-output.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockTypes, store as blocksStore } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+import { useContext, useMemo } from '@wordpress/element';
+import { generateGlobalStyles } from '@wordpress/global-styles-engine';
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../store';
+import { unlock } from '../lock-unlock';
+
+const { GlobalStylesContext, useGlobalSetting } = unlock(
+	blockEditorPrivateApis
+);
+
+/**
+ * Returns the global styles output based on the provided global styles config.
+ *
+ * @param {Object}  mergedConfig       The merged global styles config.
+ * @param {boolean} disableRootPadding Disable root padding styles.
+ *
+ * @return {Array} Array of stylesheets and settings.
+ */
+export function useGlobalStylesOutputWithConfig(
```

Comment:
Won't this ever be needed outside of edit-site? I'm thinking that strictly speaking things like the style book shouldn't really be a part of the site editor, but sit at the admin level instead.